disnake/interactions/base.py

```diff
@@ -530,6 +532,18 @@ async def send_message(
         InteractionResponded
             This interaction has already been responded to before.
         """
+        if allowed_mentions is not None:
+            if state.allowed_mentions is not None:
+                allowed_mentions = state.allowed_mentions.merge(allowed_mentions).to_dict()
+            else:
+                allowed_mentions = allowed_mentions.to_dict()
+        else:
+            allowed_mentions = state.allowed_mentions and state.allowed_mentions.to_dict()
```

Comment:
Usage of  `and` may not be exactly clear at first glance. We should keep this only if it's already used elsewhere in the project.